concepts/match-basics/about.md

```diff
@@ -1 +1,121 @@
 # Match Basics
+
+The `match` expression allows for easy comparison of a value against a series of patterns and afterwards run the code based on which pattern matches. Literal values, variable names, wildcards, and many other things are what makes up a pattern.
+
+The examples below show the `match` syntax:
+
+```rust
+match enum_var {
+    variant_0(a, b, c) => {/* code */}
+    variant_1(_) => {/* code */}
+    ...
+    variant_k(a, c) => {/* code */}
+}
+
+match felt_var {
+    0 => { /* code */ }
+    _ => { /* code */ }
+}
+```
+
+The `variant_0`, `variant_1`, `...`, and `variant_k` are all variants of an enum data type. Where `enum_var` could be any instance of the enum, after matching, executes the code in the curly braces. While the second example is the usage of the match patterning with a `felt_var` felt252 with arm patterns of value `0` and wildcard, `_`.
+
+Following the format above, we can write a function that takes in an unknown color parameter and return its true color.
+
+```rust
+enum Color {
+    Red,
+    Yellow,
+    Green,
+}
+
+fn return_color_type(color: Color) -> felt252 {
+    match color {
+        Color:Red => println!("The color is Red"),`
+        Color:Yellow => println!("The color is Yellow"),
+        Color:Green => println!("The color is Green"),
+    }
+}
+```
+
+The `return_color_type` function prints the color of whichever unknown parameter was provided. Given the unknown `color` value, it would have several patterns that involves each of the variants to match. The arm of code to execute for each pattern is a one liner `println!("The color is ${color}")` of whatever the value match. To handle mutiple lines of code, the arm of code should be wrapped into a curly braces `{}`.
```

Comment:
```suggestion
To handle multiple lines of code, the `match` arm code should be wrapped in curly braces `{}`:
```
1. Instead of explaining what would happen, provide a usage example below the match pattern function, like:
```rust
fn return_color_type(...
...
}

let color = Color::Yellow;
print_color_type(color); // prints "The color is Yellow"
```
2. Provide a simple example of the multi-line arm code (again with colors).